### ISSUE TITLE
false positive invariant_booleans on nested if

### ISSUE THREAD

**Issue by itsnuba** [dart-lang/linter#1718]
_invariant_booleans_ fired when im using if statement with or operation then using the same statement with and condition

**To Reproduce**
```
const a = true;
const b = false;
var text = "";
if (a || b) {
  text += "success! also : ";
  if (a && b) {
    text += "both value is true";
  } else if (a) {
    text += "only a is true";
  } else {
    text += "only b is true";
  }
} else {
  text += "error!";
}
print(text);
```
`a && b` is detected as _invariant_booleans_

**Expected behavior**
_invariant_booleans_ should not be fired, because the condition is not redundant with previous condition

**Additional context**
when the inner statement is only if, without else / else if, _invariant_booleans_ isnt fired
```
const a = true;
const b = false;
var text = "";
if (a || b) {
  text += "success! also : ";
  if (a && b) {
    text += "both value is true";
  }
} else {
  text += "error!";
}
print(text);
```

**Comment by jamesderlin** [comment]
`invariant_booleans` would be appropriate for the above code as written.  If anything, it should be complaining about the `if (a || b)` check too since `a` and `b` are `const`, and therefore all of the conditions are known at compilation time.  Currently (using Dart SDK 2.16.2), I don't get `invariant_booleans` warnings but do get `dead_code` warnings, which is even better.

I do get an `invariant_booleans` warning on the `if (a && b)` check with:

```dart
bool getTrue() => true;
bool getFalse() => false;

void main() {
  var a = getTrue();
  var b = getFalse();
  var text = "";
  if (a || b) {
    text += "success! also : ";
    if (a && b) {
      text += "both value is true";
    } else if (a) {
      text += "only a is true";
    } else {
      text += "only b is true";
    }
  } else {
    text += "error!";
  }
  print(text);
}
```

**Comment by srawlins** [comment]
We've deprecated `invariant_booleans` and will not be fixing correctness bugs going forward.
